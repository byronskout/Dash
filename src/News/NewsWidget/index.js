import React, { Component } from 'react';
import axios from 'axios';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import YoutubeContainer from '../../YoutubeContainer';
import Scrollbar from 'react-scrollbars-custom';

import NewsItem from '../NewsItem';

import 'react-select/dist/react-select.css';
import './NewsWidget.css';

class NewsWidget extends Component {
	constructor(props) {
		super(props);
		this.state = {
			articles: [],
			sources: [],
			currentPage: 1,
			maxPages: 1,
			selectedSource: ''
		}
	}

	handleShowMore() {
		if (this.state.currentPage < this.state.maxPages) {
			this.setState(prevState => ({
				currentPage: prevState.currentPage + 1
			}), () => this.getNews());
		}
	}

	handleChangeSource(selectedOption) {
		this.setState({ currentPage: 1, selectedSource: selectedOption == null ? '' : selectedOption.value, articles: [] }, () => this.getNews());
	}

	getNews() {
		let newsUrl;
		if (this.state.selectedSource === '') {
			newsUrl = `https://newsapi.org/v2/top-headlines?country=gb&pageSize=5&page=${this.state.currentPage}&apiKey=3988f8bf602e4bd0b0f865d02a61ebd3`;
		} else {
			newsUrl = `https://newsapi.org/v2/top-headlines?pageSize=5&page=${this.state.currentPage}&sources=${this.state.selectedSource}&apiKey=3988f8bf602e4bd0b0f865d02a61ebd3`;
		}
		axios.get(newsUrl)
			.then(res => {
				for (let article of res.data.articles) {
					this.setState(prevState => ({
						articles: [...prevState.articles, article]
					}));
				}
				this.setState({ maxPages: Math.ceil(res.data.totalResults / 5) });
			})
			.catch(err => console.log(err));
	}

	getSources() {
		axios.get(`https://newsapi.org/v2/sources?country=gb&apiKey=a3988f8bf602e4bd0b0f865d02a61ebd3`)
			.then(res => this.setState({ sources: res.data.sources }))
			.catch(err => console.log(err));
	}

	componentWillMount() {
		this.getNews();
		this.getSources();
	}

	render() {
		let articles;
		let sources;

		if (this.state.articles) {
			articles = this.state.articles.map(article => {
				return (
					<NewsItem key={article.title} article={article} />
				);
			});
		}

		if (this.state.sources) {
			sources = this.state.sources.map(source => {
				return {
					value: source.id,
					label: source.name
				};
			});
		}

		return (
			<div className="newsFeed">
				<div className="newsFeed__heading d-flex flex-row align-items-center justify-content-between">

				</div>
				<YoutubeContainer>
				<Scrollbar style={{ width: 450, height: 450 }}>
				{articles.length > 0 ? (
					<ul className="newsFeed__list">
						<ReactCSSTransitionGroup
							transitionName="newsItemsTransition"
							transitionEnterTimeout={500}
							transitionLeaveTimeout={300}>
							{articles}
						</ReactCSSTransitionGroup>
					</ul>
				) : (
					<p className="newsFeed__empty">No news articles were found.</p>
				)}
				<button
					className="newsFeed__more btn btn-primary"
					onClick={this.handleShowMore.bind(this)}>
					{this.state.currentPage === this.state.maxPages ? 'No More News' : 'Show More'}</button>
					</Scrollbar>
					</YoutubeContainer>
			</div>
		)
	}
}

export default NewsWidget;
