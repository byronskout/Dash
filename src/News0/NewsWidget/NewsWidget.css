.newsFeed {
	padding: 2rem;
	margin-top: 2rem;
	margin-bottom: 2rem;
	background-color: #f7f7f7;
}

.newsFeed__heading {
	margin-bottom: 2rem;
}

.newsFeed__title {
	padding-bottom: 0.25rem;
	margin: 0;
	font-size: 1.6rem;
	font-weight: 700;
	line-height: 1;
	color: #dc3f8e;
	border-bottom: 4px solid #dc3f8e;
}

.newsFeed__sources {
	width: 11rem;
}

.newsFeed__sources .Select-control {
	height: auto;
	background-color: transparent;
	border: 2px solid #949494;
	border-radius: 5px;
}

.newsFeed__sources.is-focused:not(.is-open) .Select-control,
.newsFeed__sources.is-open .Select-control {
	border-color: #dc3f8e;
	box-shadow: 0 0 0 transparent;
}

.newsFeed__sources .Select-placeholder {
	font-size: 1rem;
	color: #949494;
}

.newsFeed__sources .Select-option.is-focused {
	background-color: #ffebf9;
}

.newsFeed__list {
	padding: 0;
	margin: 0 0 2rem;
	list-style: none;
}

.btn.btn-primary {
	background-color: transparent;
	border: 3px solid #dc3f8e;
	font-size: 1rem;
	font-weight: 700;
	color: #dc3f8e;
	border-radius: 0;
	transition: all .2s ease-in-out;
}

.btn.btn-primary:hover,
.btn.btn-primary:focus,
.btn.btn-primary:not(:disabled):not(.disabled):active {
	background-color: #dc3f8e;
	border-color: #dc3f8e;
	color: #fff;
	box-shadow: 0 0 0 transparent;
}

.newsFeed__sponsor {
	margin-top: 2rem;
}

.newsItemsTransition-enter {
	opacity: 0.01;
}

.newsItemsTransition-enter.newsItemsTransition-enter-active {
	opacity: 1;
	transition: opacity 500ms ease-in;
}

.newsItemsTransition-leave {
	opacity: 1;
}

.newsItemsTransition-leave.newsItemsTransition-leave-active {
	opacity: 0.01;
	transition: opacity 300ms ease-in;
}
