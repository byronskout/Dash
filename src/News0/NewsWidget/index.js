import React, { Component } from 'react';
import axios from 'axios';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

import NewsItem from '../NewsItem';
import Select from 'react-select';

import 'react-select/dist/react-select.css';
import './NewsWidget.css';

class NewsWidget extends Component {
	constructor(props) {
		super(props);
		this.state = {
			articles: [],
			sources: [],
			currentPage: 1,
			maxPages: 1,
			selectedSource: ''
		}
	}

	handleShowMore() {
		if (this.state.currentPage < this.state.maxPages) {
			this.setState(prevState => ({
				currentPage: prevState.currentPage + 1
			}), () => this.getNews());
		}
	}

	handleChangeSource(selectedOption) {
		this.setState({ currentPage: 1, selectedSource: selectedOption == null ? '' : selectedOption.value, articles: [] }, () => this.getNews());
	}

	getNews() {
		let newsUrl;
		if (this.state.selectedSource === '') {
			newsUrl = `https://newsapi.org/v2/top-headlines?country=gb&pageSize=5&page=${this.state.currentPage}&apiKey=17fcb50c25f244a59d6a87fda4730bef`;
		} else {
			newsUrl = `https://newsapi.org/v2/top-headlines?pageSize=5&page=${this.state.currentPage}&sources=${this.state.selectedSource}&apiKey=17fcb50c25f244a59d6a87fda4730bef`;
		}
		axios.get(newsUrl)
			.then(res => {
				for (let article of res.data.articles) {
					this.setState(prevState => ({
						articles: [...prevState.articles, article]
					}));
				}
				this.setState({ maxPages: Math.ceil(res.data.totalResults / 5) });
			})
			.catch(err => console.log(err));
	}

	getSources() {
		axios.get(`https://newsapi.org/v2/sources?country=gb&apiKey=17fcb50c25f244a59d6a87fda4730bef`)
			.then(res => this.setState({ sources: res.data.sources }))
			.catch(err => console.log(err));
	}

	componentWillMount() {
		this.getNews();
		this.getSources();
	}

	render() {
		let articles;
		let sources;

		if (this.state.articles) {
			articles = this.state.articles.map(article => {
				return (
					<NewsItem key={article.title} article={article} />
				);
			});
		}

		if (this.state.sources) {
			sources = this.state.sources.map(source => {
				return {
					value: source.id,
					label: source.name
				};
			});
		}

		return (
			<div className="newsFeed">
				<div className="newsFeed__heading d-flex flex-row align-items-center justify-content-between">
					<h3 className="newsFeed__title">News</h3>
					<Select
						className="newsFeed__sources"
						placeholder="Filter By Source"
						value={this.state.selectedSource}
						options={sources.length > 0 ? sources : null}
						onChange={this.handleChangeSource.bind(this)}
					/>
				</div>
				{articles.length > 0 ? (
					<ul className="newsFeed__list">
						<ReactCSSTransitionGroup
							transitionName="newsItemsTransition"
							transitionEnterTimeout={500}
							transitionLeaveTimeout={300}>
							{articles}
						</ReactCSSTransitionGroup>
					</ul>
				) : (
					<p className="newsFeed__empty">No news articles were found.</p>
				)}
				<button
					className="newsFeed__more btn btn-primary"
					onClick={this.handleShowMore.bind(this)}>
					{this.state.currentPage === this.state.maxPages ? 'No More News' : 'Show More'}</button>
				<p className="newsFeed__sponsor">Powered by <a href="https://newsapi.org/" target="_blank" rel="noopener noreferrer">News API</a>.</p>
			</div>
		)
	}
}

export default NewsWidget;
