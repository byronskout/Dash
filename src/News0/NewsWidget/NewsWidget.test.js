import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import axios from 'axios';
import moxios from 'moxios';

import NewsWidget from './index';

describe(NewsWidget, () => {
	configure({ adapter: new Adapter() });

	beforeEach(() => {
		moxios.install(axios);
	});

	afterEach(() => {
		moxios.uninstall(axios);
	});

	it('should render the component correctly and match snapshot', () => {
		const component = renderer.create(<NewsWidget />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
