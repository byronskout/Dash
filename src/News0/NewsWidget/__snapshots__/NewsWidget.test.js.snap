// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class NewsWidget extends _react.Component {
	constructor(props) {
		super(props);
		this.state = {
			articles: [],
			sources: [],
			currentPage: 1,
			maxPages: 1,
			selectedSource: '' };

	}

	handleShowMore() {
		if (this.state.currentPage < this.state.maxPages) {
			this.setState(prevState => ({
				currentPage: prevState.currentPage + 1 }),
			() => this.getNews());
		}
	}

	handleChangeSource(selectedOption) {
		this.setState({ currentPage: 1, selectedSource: selectedOption == null ? '' : selectedOption.value, articles: [] }, () => this.getNews());
	}

	getNews() {
		let newsUrl;
		if (this.state.selectedSource === '') {
			newsUrl = \`https://newsapi.org/v2/top-headlines?country=gb&pageSize=5&page=\${this.state.currentPage}&apiKey=17fcb50c25f244a59d6a87fda4730bef\`;
		} else {
			newsUrl = \`https://newsapi.org/v2/top-headlines?pageSize=5&page=\${this.state.currentPage}&sources=\${this.state.selectedSource}&apiKey=17fcb50c25f244a59d6a87fda4730bef\`;
		}
		_axios2.default.get(newsUrl).
		then(res => {
			for (let article of res.data.articles) {
				this.setState(prevState => ({
					articles: [...prevState.articles, article] }));

			}
			this.setState({ maxPages: Math.ceil(res.data.totalResults / 5) });
		}).
		catch(err => console.log(err));
	}

	getSources() {
		_axios2.default.get(\`https://newsapi.org/v2/sources?country=gb&apiKey=17fcb50c25f244a59d6a87fda4730bef\`).
		then(res => this.setState({ sources: res.data.sources })).
		catch(err => console.log(err));
	}

	componentWillMount() {
		this.getNews();
		this.getSources();
	}

	render() {
		let articles;
		let sources;

		if (this.state.articles) {
			articles = this.state.articles.map(article => {
				return (
					_react2.default.createElement(_NewsItem2.default, { key: article.title, article: article, __source: { fileName: _jsxFileName, lineNumber: 72 }, __self: this }));

			});
		}

		if (this.state.sources) {
			sources = this.state.sources.map(source => {
				return {
					value: source.id,
					label: source.name };

			});
		}

		return (
			_react2.default.createElement('div', { className: 'newsFeed', __source: { fileName: _jsxFileName, lineNumber: 87 }, __self: this },
				_react2.default.createElement('div', { className: 'newsFeed__heading d-flex flex-row align-items-center justify-content-between', __source: { fileName: _jsxFileName, lineNumber: 88 }, __self: this },
					_react2.default.createElement('h3', { className: 'newsFeed__title', __source: { fileName: _jsxFileName, lineNumber: 89 }, __self: this }, 'News'),
					_react2.default.createElement(_reactSelect2.default, {
						className: 'newsFeed__sources',
						placeholder: 'Filter By Source',
						value: this.state.selectedSource,
						options: sources.length > 0 ? sources : null,
						onChange: this.handleChangeSource.bind(this), __source: { fileName: _jsxFileName, lineNumber: 90 }, __self: this })),


				articles.length > 0 ?
				_react2.default.createElement('ul', { className: 'newsFeed__list', __source: { fileName: _jsxFileName, lineNumber: 99 }, __self: this },
					_react2.default.createElement(_reactAddonsCssTransitionGroup2.default, {
							transitionName: 'newsItemsTransition',
							transitionEnterTimeout: 500,
							transitionLeaveTimeout: 300, __source: { fileName: _jsxFileName, lineNumber: 100 }, __self: this },
						articles)) :



				_react2.default.createElement('p', { className: 'newsFeed__empty', __source: { fileName: _jsxFileName, lineNumber: 108 }, __self: this }, 'No news articles were found.'),

				_react2.default.createElement('button', {
						className: 'newsFeed__more btn btn-primary',
						onClick: this.handleShowMore.bind(this), __source: { fileName: _jsxFileName, lineNumber: 110 }, __self: this },
					this.state.currentPage === this.state.maxPages ? 'No More News' : 'Show More'),
				_react2.default.createElement('p', { className: 'newsFeed__sponsor', __source: { fileName: _jsxFileName, lineNumber: 114 }, __self: this }, 'Powered by ', _react2.default.createElement('a', { href: 'https://newsapi.org/', target: '_blank', rel: 'noopener noreferrer', __source: { fileName: _jsxFileName, lineNumber: 114 }, __self: this }, 'News API'), '.')));


	}} should render the component correctly and match snapshot 1`] = `
<div
  className="newsFeed"
>
  <div
    className="newsFeed__heading d-flex flex-row align-items-center justify-content-between"
  >
    <h3
      className="newsFeed__title"
    >
      News
    </h3>
    <div
      className="Select newsFeed__sources is-clearable is-searchable Select--single"
      style={undefined}
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-2--value"
        >
          <div
            className="Select-placeholder"
          >
            Filter By Source
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-2--value"
              aria-describedby={undefined}
              aria-expanded="false"
              aria-haspopup="false"
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-owns=""
              className={undefined}
              id={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              tabIndex={undefined}
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
  <p
    className="newsFeed__empty"
  >
    No news articles were found.
  </p>
  <button
    className="newsFeed__more btn btn-primary"
    onClick={[Function]}
  >
    No More News
  </button>
  <p
    className="newsFeed__sponsor"
  >
    Powered by 
    <a
      href="https://newsapi.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      News API
    </a>
    .
  </p>
</div>
`;
